### PR TITLE
Complete Documentation

### PR DESCRIPTION
**Pull request by greatsharma**
This PR completes the missing documentation.

### CODE REVIEW

File: src/mlpack/methods/logistic_regression/logistic_regression_function.hpp [hunk 1]
```diff
@@ -142,6 +142,11 @@ class LogisticRegressionFunction
   double EvaluateWithGradient(const arma::mat& parameters,
                               GradType& gradient) const;
 
+  /**
+   * Evaluate the objective function and gradient of the logistic regression
+   * log-likelihood function simultaneously with the given parameters, for
+   * the given batch size from a given point the in dataset.
+   */
```
Comment (zoq) [inline src/mlpack/methods/logistic_regression/logistic_regression_function.hpp]: Looks like there is no description for the constructor as well, do you like to fix that as well? Don't feel obligated. Also, not sure why it's not done here, but usually we provide a description for each parameter, which I think is essential for the documentation processing step. Let me know what you think.

Comment (greatsharma) [reply src/mlpack/methods/logistic_regression/logistic_regression_function.hpp]: @zoq I included the documentation for the constructors and also did minor documentation fixUps.

File: src/mlpack/methods/logistic_regression/logistic_regression_function.hpp [hunk 2]
```diff
@@ -28,10 +28,25 @@ template<typename MatType = arma::mat>
...
   LogisticRegressionFunction(const MatType& predictors,
                              const arma::Row<size_t>& responses,
                              const double lambda = 0);
 
+  /**
+   * Creates the LogisticRegressionFunction with initialPoint.
+   *
+   * @param predictors X, matrix of data points.
+   * @param responses y, the measured data for each point in X.
+   * @param initialPoint from which to start the optimization.
```
Comment (ShikharJ) [inline src/mlpack/methods/logistic_regression/logistic_regression_function.hpp]: It should be `Point from which ...`

File: src/mlpack/methods/logistic_regression/logistic_regression_function.hpp [hunk 3]
```diff
@@ -28,10 +28,25 @@ template<typename MatType = arma::mat>
 class LogisticRegressionFunction
 {
  public:
+  /**
+   * Creates the LogisticRegressionFunction.
+   *
+   * @param predictors X, matrix of data points.
+   * @param responses y, the measured data for each point in X.
```
Comment (ShikharJ) [inline src/mlpack/methods/logistic_regression/logistic_regression_function.hpp]: I'd rather not give variable notations to the `predictors` and `responses`. Please make it something like
```
@param predictors The matrix of data points.
@param responses The measured data for each point in predictors.
```

File: src/mlpack/methods/logistic_regression/logistic_regression_function.hpp [hunk 4]
```diff
@@ -28,10 +28,25 @@ template<typename MatType = arma::mat>
...
+   */
   LogisticRegressionFunction(const MatType& predictors,
                              const arma::Row<size_t>& responses,
                              const double lambda = 0);
 
+  /**
+   * Creates the LogisticRegressionFunction with initialPoint.
+   *
+   * @param predictors X, matrix of data points.
+   * @param responses y, the measured data for each point in X.
```
Comment (ShikharJ) [inline src/mlpack/methods/logistic_regression/logistic_regression_function.hpp]: Same here as above.

Comment (greatsharma) [reply src/mlpack/methods/logistic_regression/logistic_regression_function.hpp]: @ShikharJ  I did the changes suggested by you

**Comment by ShikharJ** [comment]
Thanks @greatsharma for the contribution!

**Pull request merged**